### PR TITLE
Update sample files to use universal headers

### PR DESCRIPTION
**Pull request by momuno**

### CODE REVIEW

**Review by antkmsft — APPROVED** [review]
LGTM, I have some suggestions about related issues, but none are blocking this PR, this PR is good as it is, but if you want to address my other feedback in this PR, it is also fine. It is also fine to disagree with my suggestions and not do any of it.

File: sdk/samples/iot/pnp/pnp_mqtt_message.c [hunk 1]
```diff
@@ -5,15 +5,12 @@
 
 #include <iot_sample_common.h>
```
[the remark this answers is not in the document]
Comment (momuno) [reply sdk/samples/iot/pnp/pnp_mqtt_message.c:6]: We have used the "" for files within the same directory or sub directory.  The <> is used for files outside of the immediate directory tree which would be in the include path list.  It is compiler dependent, but this was our reasoning.
https://stackoverflow.com/questions/3162030/difference-between-angle-bracket-and-double-quotes-while-including-heade

File: sdk/samples/iot/pnp/pnp_device_info_component.c [hunk 2]
```diff
@@ -5,12 +5,9 @@
 
 #include <iot_sample_common.h>
 
-#include <stddef.h>
+#include <azure/az_core.h>
```
[the remark this answers is not in the document]
Comment (momuno) [reply sdk/samples/iot/pnp/pnp_device_info_component.c:8]: That makes sense.  So in this order?

<azure/az_core.h etc>

<std lib stuff>

<iot sample common>

"local directory headers"

[the remark this answers is not in the document]
Comment (antkmsft) [reply sdk/samples/iot/pnp/pnp_device_info_component.c:4]: IoT sample:
```
<stdlib>

<other libs>

<azure sdk>

<iot sample>
"iot sample local"
```

Azure SDK source code:
```
<azure sdk public>

<azure sdk internal>
"azure sdk private"

<stdlib>

<other libs>
```

[the remark this answers is not in the document]
Comment (momuno) [reply sdk/samples/iot/pnp/pnp_device_info_component.c:4]: why the ordering difference between the two? std libs and other libs are at the top in IoT Sample, but the bottom in sdk source code

**Pull request merged**